From 7e5b7bcdeb6c4853c45ca16f851fc0dccce046cc Mon Sep 17 00:00:00 2001
From: Matthew Quijano <mq022002@gmail.com>
Date: Sun, 10 Mar 2024 13:51:00 -0400
Subject: [PATCH 07/42] Refactor: Finalized structure for Firestore database

- Adjust quiz titles and deprecated clean_quiz_title function. Might use later, but it is probably not needed considering what we can do with chapterType on the frontend to explicitly tell the user that this is a quiz.
---
 data/topics2.json   | 184 ++++++++++++++++++++++++++++++++++++++++++++
 firebase_convert.py |  10 +--
 2 files changed, 189 insertions(+), 5 deletions(-)
 create mode 100644 data/topics2.json

diff --git a/data/topics2.json b/data/topics2.json
new file mode 100644
index 0000000..28d9ebb
--- /dev/null
+++ b/data/topics2.json
@@ -0,0 +1,184 @@
+{
+  "topics": [
+    {
+      "topicTitle": "Two-Factor Authentication",
+      "topicDescription": "A course that explores the concept of Two-Factor Authentication.",
+      "order": 1,
+      "chapters": [
+        {
+          "chapterTitle": "Introduction to 2FA",
+          "chapterDescription": "An overview of the concept of Two-Factor Authentication.",
+          "chapterType": "text",
+          "controlGroupContent": "An introduction to Two-Factor Authentication (2FA) offers a fundamental understanding of this essential security measure in today's digital landscape. Two-Factor Authentication enhances account security by requiring users to provide two forms of identification before granting access, typically something they know (like a password) and something they have (such as a smartphone or hardware token). This introductory overview delves into the significance of 2FA in safeguarding sensitive information from unauthorized access and mitigating the risks associated with password breaches and phishing attacks. By providing a layered defense mechanism, 2FA adds an extra layer of protection to online accounts, bolstering security measures for individuals and organizations alike in an increasingly interconnected world.",
+          "controlGroupImageURLs": [
+            "https://assets.f-secure.com/i/illustrations/what-is-two-factor-authentication.png"
+          ],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 1
+        },
+        {
+          "chapterTitle": "Understanding 2FA",
+          "chapterDescription": "A quiz to test your knowledge of Two-Factor Authentication.",
+          "chapterType": "assessment",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 2
+        },
+        {
+          "chapterTitle": "Implementing 2FA Solutions",
+          "chapterDescription": "Guidelines for implementing Two-Factor Authentication solutions.",
+          "chapterType": "text",
+          "controlGroupContent": "https://www.youtube.com/embed/AMOtB7XkTT4",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 3
+        }
+      ]
+    },
+    {
+      "topicTitle": "Phishing",
+      "topicDescription": "A comprehensive guide to understanding and preventing phishing attacks.",
+      "order": 2,
+      "chapters": [
+        {
+          "chapterTitle": "Introduction to Phishing",
+          "chapterDescription": "An overview of the concept of phishing and its impact.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 1
+        },
+        {
+          "chapterTitle": "Assessing Phishing Knowledge",
+          "chapterDescription": "A quiz to test your knowledge of phishing attacks.",
+          "chapterType": "assessment",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 2
+        },
+        {
+          "chapterTitle": "Recognizing and Reporting Phishing Attempts",
+          "chapterDescription": "Strategies for recognizing and reporting phishing attempts.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 3
+        }
+      ]
+    },
+    {
+      "topicTitle": "Password Security",
+      "topicDescription": "Learn about the importance of strong passwords and how to create them.",
+      "order": 3,
+      "chapters": [
+        {
+          "chapterTitle": "Introduction to Password Security",
+          "chapterDescription": "An overview of the importance of strong passwords.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 1
+        },
+        {
+          "chapterTitle": "Creating Strong Passwords",
+          "chapterDescription": "Tips and techniques for creating strong and secure passwords.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 2
+        },
+        {
+          "chapterTitle": "Password Security",
+          "chapterDescription": "A quiz to test your knowledge of password security best practices.",
+          "chapterType": "assessment",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 3
+        },
+        {
+          "chapterTitle": "Implementing Password Management Solutions",
+          "chapterDescription": "Guidelines for implementing password management solutions.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 4
+        }
+      ]
+    },
+    {
+      "topicTitle": "Online Privacy",
+      "topicDescription": "A course that delves into the concept of online privacy and how to protect it.",
+      "order": 4,
+      "chapters": [
+        {
+          "chapterTitle": "Introduction to Online Privacy",
+          "chapterDescription": "An overview of the concept of online privacy and its importance.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 1
+        },
+        {
+          "chapterTitle": "Protecting Your Online Privacy",
+          "chapterDescription": "Best practices for protecting your online privacy and personal information.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 2
+        },
+        {
+          "chapterTitle": "Evaluating Understanding of Online Privacy",
+          "chapterDescription": "A quiz to test your knowledge of online privacy.",
+          "chapterType": "assessment",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 3
+        },
+        {
+          "chapterTitle": "Privacy Settings and Tools",
+          "chapterDescription": "A guide to adjusting privacy settings and using privacy tools on various platforms.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 4
+        },
+        {
+          "chapterTitle": "Implementing Privacy Enhancements",
+          "chapterDescription": "Guidelines for implementing privacy enhancements.",
+          "chapterType": "text",
+          "controlGroupContent": "",
+          "controlGroupImageURLs": [],
+          "experimentalGroupContent": "",
+          "experimentalGroupImageURLs": [],
+          "order": 5
+        }
+      ]
+    }
+  ]
+}
diff --git a/firebase_convert.py b/firebase_convert.py
index c07569b..403c3bc 100644
--- a/firebase_convert.py
+++ b/firebase_convert.py
@@ -21,13 +21,13 @@ def to_camel_case(text):
     return components[0] + "".join(x.title() for x in components[1:])
 
 
-def clean_quiz_title(title):
-    if "Quiz" in title:
-        return title.replace(":", "")
-    return title
+# def clean_quiz_title(title):
+#     if "Quiz" in title:
+#         return title.replace(":", "")
+#     return title
 
 
-with open("data/topics.json") as f:
+with open("data/topics2.json") as f:
     data = json.load(f)
 
 for topic_details in data["topics"]:
-- 
2.38.1.windows.1

